### DISCUSSION TITLE
Getting two values from one request

### DISCUSSION THREAD

**Discussion by WoodsterDK** [dbosoft/YaNco#231] [Q&A] [answered]
Hi,

I have a RFC that returns two values, but I cannot seem to figure out how to get two values from the output section.

It seems only one .GetField<string>("VALUE1") is possible.......

was thinking something like:

.GetField<string>("VALUE1")
.GetField<string>("VALUE2")

How is that managed?

**Comment by fw2568** [comment]
Hello,

it is possible, as you can simply perform multiple mappings within one mapping function. See an enhanced version of the company code sample from readme:

``` csharp
using (var context = new RfcContext(connFunc))
{
    await context.CallFunction("BAPI_COMPANYCODE_GETDETAIL",
        Input: f => f
            .SetField("COMPANYCODEID", "1000"),
        Output: f => f
            .MapStructure("COMPANYCODE_DETAIL", s =>
                from code in s.GetField<string>("COMP")
                from name in s.GetField<string>("COMP_NAME")
                select new { code, name }
            ))

            .Match(r => Console.WriteLine($"Result: {r.code} - {r.name}"),
                l => Console.WriteLine($"Error: {l.Message}"));

}
```

So here both the code and the name are assigned to an anonymous type (it could also be a defined type) and returned. The key to understanding this is that you can map the output to any type.

Hope this helps,

Best Regards,
Frank

**Reply by WoodsterDK** [reply]
Hi Frank,

Thanks for getting back.
I did see the example, but that mapping takes into account that data is either in a struct or table, but for the RFC I have to use, data is returned as two single values.... how is that done?

**Comment by fw2568** [comment, marked as answer]
same logic, but directly on output mapping function:

``` csharp
 Output: f => 
  from field1 in f.GetField<string>("FIELD1")
  from field2 in f.GetField<string>("FIELD2")
  select new { field1, field2 }
            
```

**Reply by WoodsterDK** [reply]
Once again thanks for quick response.

I actually tried something like that, but it had issues. Must have been a typo, will test later.

Thanks...
